### ISSUE TITLE
Menu - right click context menu can fail to expand in certain circumstances.

### ISSUE THREAD

**Issue by smurphos** [linuxmint/Cinnamon#7427]
```
 * Cinnamon version 3.6.7
 * Distribution - Mint 18.3
 *  Card: Intel 2nd Generation Core Processor Family Integrated Graphics Controller
      bus-ID: 00:02.0
      Display Server: X.Org 1.18.4 drivers: (unloaded: fbdev,vesa)
      Resolution: 1366x768@60.03hz
      GLX Renderer: Mesa DRI Intel Sandybridge Mobile
      GLX Version: 3.0 Mesa 17.2.8 Direct Rendering: Yes

 *  64 bit
 ```

**Issue**

Right click context menu can fail to expand in certain combinations of theme and screen resolution.

**Steps to reproduce**

This bug seems to be triggered if system or theme characteristics result in a menu with a height close to or exceeding the monitor height. E.g. setting text-scaling to a value above 1, having a theme with a large font size set, or a theme with relatively large padding values for menu-xxx-button styles.

The screen-shots below were taken on a 1366x768 monitor. Picture one is with an unmodified default theme and text scaling set to 2.3. Picture 2 is with text-scaling 1.0 but the default theme modified so that all menu-xxx-button styles have padding of 18px.

![screenshot-screen-2018-04-02-072558](https://user-images.githubusercontent.com/29017677/38185882-134a73d0-3648-11e8-8073-a27d6980c9f2.png)
![screenshot-screen-2018-04-02-072947](https://user-images.githubusercontent.com/29017677/38185887-19ec56ae-3648-11e8-9c3f-d7ff1fbb2f65.png)

**Expected behaviour**

**Other information**

**Comment by smurphos** [comment]
Having a min-height set for .popup-menu-item widens the circumstances in which this will manifest.

**Comment by CottonEaster** [comment]
@smurphos 

> Having a min-height set for .popup-menu-item widens the circumstances in which this will manifest - edit to add - actually it's a foolproof way of manifesting the issue.

Does this mean you are in a position to suggest a workaround, please?

**Comment by smurphos** [comment]
@CottonEaster 
> 
> Does this mean you are in a position to suggest a workaround, please?

Yep - make sure you change whatever factor is it that makes a menu that is close to to the monitor height in size. Could be too many favorites in the stock menu, font-scaling, the theme itself....

**Comment by CottonEaster** [comment]
smurphos: thanks, but for some reason (I don't know what) I do not have the problem at the moment; and any way you look at it, this is a bug, I think. For while admittedly a theme could be response, Cinnamon should, ideally anyway, have a way of checking for and coping with such problems.

**Comment by LinuxOnTheDesktop** [comment]
Perhaps it is worth reporting the following.

I have this problem of the absent context-menu and even on the alpha of Mint 19.2. I tried three screen resolutions (on the same computer) and have the problem in all of them. Disabling the left pane did not help. However, disabling HiDPI - which cannot be a permanent solution for me - did solve the problem. So: _it seems to be a DPI bug_. If that is the sole cause, it could bite a lot of people.

**Comment by LinuxOnTheDesktop** [comment]
Once again I have this problem and I find that the workaround suggested by @Smurphos does work. (Thanks, Smurpho!) That workaround is: find a way of making the menu height - the height of the main window pane - at least some 20% less than the size of the monitor. In my case I managed to implement the workaroubd by disabling both 'show category items' and 'show bookmarks and places'.
